### PR TITLE
Adding `open_datatree` backend-specific keyword arguments

### PR DESCRIPTION
**Pull request by aladinor**
- [x] Closes ##9135
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: doc/whats-new.rst [hunk 1]
```diff
@@ -22,6 +22,17 @@ v2024.06.1 (unreleased)
...
+Performance
+~~~~~~~~~~~
+
+- Small optimization to the netCDF4 and h5netcdf backends (:issue:`9058`, :pull:`9067`).
+  By `Deepak Cherian <https://github.com/dcherian>`_.
+- Small optimizations to help reduce indexing speed of datasets (:pull:`9002`).
+  By `Mark Harfouche <https://github.com/hmaarrfk>`_.
+- Performance improvement in `open_datatree` method for Zarr, netCDF4 and h5netcdf backends (:issue:`8994`, :pull:`9014`).
+  By `Alfonso Ladino <https://github.com/aladinor>`_.
+
```
Comment (dcherian) [inline doc/whats-new.rst:35]:
```suggestion
```

**Comment by keewis** [comment]
you might want to rebase / cherry-pick the two commits for this PR onto `main`, so there's only those two commits to check in this PR. If you don't have any idea how to do that, I can do it for you.

**Comment by keewis** [comment]
so unfortunately I can't push to your repository, because it is a fork of someone else's fork (and I'm not a maintainer on that fork). To fix that, you can either [request](https://support.github.com/contact?subject=Reroute%20a%20Fork&tags=rr-forks) someone from github to "reroute" your fork, or you can pull all your branches to local (I'm sure you did already), delete the repository and fork again. I'm not sure what would happen to this PR, though.

**Comment by keewis** [comment]
if you want to finish up this PR, you can also follow the steps yourself. Basically, what you do is:
```sh
git switch main
git pull
git switch open-dtree-kwargs
# rewind to main
git reset --hard HEAD~60
git merge main
# apply the two commits that were added in this PR
git cherry-pick 179b6960c6e2f49b38eda13a97d977a72bcb5990
git cherry-pick 563b5dbda4dabd253c38c9ca577015224447314f
git push --force
```

**Review by keewis — COMMENTED** [review]
looks good to me. Do we need tests for this?

**Comment by JessicaS11** [comment]
> Do we need tests for this?

Funny you should ask - it turns out this functionality wasn't tested to begin with, so @eni-awowale and I are actually working on implementing some (started at SciPy sprints and not yet opened as a draft PR, but located [here](https://github.com/JessicaS11/xarray/tree/dtree-tests)). Thanks to @santisoler for his help getting that effort started during Sprints!

**Comment by TomNicholas** [comment]
@JessicaS11 it would be great if you could push whatever you have as a PR (don't worry about perfecting it!) just so that those tests don't get lost in the ether

**Pull request merged**